third_party/move/move-prover/tests/sources/functional/verify_vector.move

```diff
@@ -146,28 +146,26 @@ module 0x42::VerifyVector {
     }
 
     // Moves all of the elements of the `other` vector into the `v` vector.
-    fun verify_append<Element>(v: &mut vector<Element>, other: vector<Element>) {
-        let o = &mut other;
-        vector::reverse(o);
+    fun verify_append<Element>(v: &mut vector<Element>, other: &mut vector<Element>) {
+        vector::reverse(other);
         while ({
             spec {
                 invariant len(v) >= len(old(v));
-                invariant len(o) <= len(other);
-                invariant len(v) + len(o) == len(old(v)) + len(other);
+                invariant len(other) <= len(old(other));
+                invariant len(v) + len(other) == len(old(v)) + len(old(other));
```

Comment:
The original invariant should not hold because `o` is a mutable reference of `other` and thus `len(o)` should be equal to `len(other)`.